l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -264,43 +237,91 @@ def getPartnerBase(self, DatiAnagrafici):  # noqa: C901
-            if DatiAnagrafici.Anagrafica.Denominazione:
-                vals["name"] = DatiAnagrafici.Anagrafica.Denominazione
+        return commercial_partner_id
+
+    def _find_partners_by_cf_vat(self, DatiAnagrafici):
+        partner_model = self.env["res.partner"]
+        cf, vat = self._define_vat_fiscalcode(DatiAnagrafici)
+
+        partners = partner_model
+        if vat:
+            domain = [("vat", "=", vat)]
+            domain = self._add_company_domain_from_attachment(domain)
+            partners = partner_model.search(domain)
+        if not partners and cf:
+            domain = [("fiscalcode", "=", cf)]
+            domain = self._add_company_domain_from_attachment(domain)
+            partners = partner_model.search(domain)
+        return partners
+
+    def _add_company_domain_from_attachment(self, domain):
+        """
+        Extends domain with company_id filter depending on
+        attachment in context
+        """
+        res_partner_rule = (
+            self.env["ir.model.data"]
+            .sudo()
+            .xmlid_to_object("base.res_partner_rule", raise_if_not_found=False)
+        )
+        if (
+            self.env.context.get("from_attachment")
+            and res_partner_rule
+            and res_partner_rule.active
+        ):
+            att = self.env.context.get("from_attachment")
+            domain.extend(
+                [
+                    "|",
+                    ("company_id", "child_of", att.company_id.id),
+                    ("company_id", "=", False),
+                ]
+            )
+        return domain
+
+    def _define_vat_fiscalcode(self, DatiAnagrafici):
+        cf = DatiAnagrafici.CodiceFiscale or False
+        vat = False
+        if DatiAnagrafici.IdFiscaleIVA:
+            id_paese = DatiAnagrafici.IdFiscaleIVA.IdPaese.upper()
+            id_codice = re.sub(r"\W+", "", DatiAnagrafici.IdFiscaleIVA.IdCodice).upper()
+            # Format Italian VAT ID to always have 11 char
+            # to avoid validation error when creating the given partner
+            if id_paese == "IT" and not id_codice.startswith("IT"):
+                vat = "IT{}".format(id_codice.rjust(11, "0")[:11])
+            # XXX maybe San Marino needs special formatting too?
+            else:
+                vat = id_codice
+        return cf, vat
+
+    def _prepare_partner_values(self, DatiAnagrafici, vat=False, cf=False):
+        # define partner vals
+        country_id = False
+        if DatiAnagrafici.IdFiscaleIVA:
+            CountryCode = DatiAnagrafici.IdFiscaleIVA.IdPaese
+            countries = self.CountryByCode(CountryCode)
+            if countries:
+                country_id = countries[0].id
+            else:
+                raise UserError(_("Country Code %s not found in system.") % CountryCode)
+        vals = {
+            "vat": vat if vat else False,
+            "fiscalcode": cf if cf else False,
```

Comment:
Se `vat` (o `cf`) sono vuoti, cosa cambia scriverci invece `False`?
Se non cambia nulla puoi rimettere com'era prima?
```suggestion
            "vat": vat,
            "fiscalcode": cf,
```